lib/npg_qc/autoqc/checks/rna_seqc.pm

```diff
@@ -0,0 +1,328 @@
+use npg_tracking::util::types;
+
+extends qw(npg_qc::autoqc::checks::check);
+
+with qw(npg_tracking::data::reference::find 
+        npg_common::roles::software_location
+        npg_tracking::data::transcriptome::find
+       );
+
+our $VERSION = '0';
+
+Readonly::Scalar our $EXT => q[bam];
+Readonly::Scalar my $RNASEQC_JAR_NAME       => q[RNA-SeQC_v1.1.8.jar];
+Readonly::Scalar my $RNASEQC_JAR_VERSION    => q[1.1.8];
+Readonly::Scalar my $RNASEQC_GTF_TTYPE_COL  => 2;
+Readonly::Scalar my $RNASEQC_GTF_DIRECTORY  => q[RNA-SeQC];
+Readonly::Scalar my $RNASEQC_OUTDIR         => q[rna_seqc];
+Readonly::Scalar my $JAVA_MAX_HEAP_SIZE     => q[4000m];
+Readonly::Scalar my $JAVA_GC_TYPE           => q[+UseSerialGC];
+Readonly::Scalar my $JAVA_USE_PERF_DATA     => q[-UsePerfData];
+
+has '+file_type' => (default => $EXT,);
+
+has '+aligner' => (default => q[fasta],);
+
+has 'java_max_heap_size'     => (is      => 'ro',
+                                 isa     => 'Str',
+                                 default => $JAVA_MAX_HEAP_SIZE,);
+
+has 'java_gc_type'           => (is      => 'ro',
+                                 isa     => 'Str',
+                                 default => $JAVA_GC_TYPE,);
+
+has 'java_use_perf_data'     => (is      => 'ro',
+                                 isa     => 'Str',
+                                 default => $JAVA_USE_PERF_DATA,);
+
+has 'java_jar_path'          => (is      => 'ro',
+                                 isa     => 'NpgCommonResolvedPathJarFile',
+                                 coerce  => 1,
+                                 default => $RNASEQC_JAR_NAME,);
+
+has 'transcript_type' => (is      => 'ro',
+                          isa     => 'Int',
+                          default => $RNASEQC_GTF_TTYPE_COL,);
+
+
+has 'alignments_in_bam' => (is         => 'ro',
+                            isa        => 'Maybe[Bool]',
+                            lazy_build => 1,);
+
+sub _build_alignments_in_bam {
+    my ($self) = @_;
+    return $self->lims->alignments_in_bam;
+}
+
+###############################################################################
+# Output path:
+#  By default, the pipeline supplies the archive/qc/rna_seqc path under the
+#  Latest_Summary directory via the qc_out parameter. This value is used to
+#  lazy-build the output path for the RNA-SeQC results within the following
+#  path: <qc_out path>/<run_position>/<run_position#tag>. qc_out remains
+#  intact so the JSON files end up stored in a non-standard location. This
+#  behaviour may change in the future if metrics are to be read and stored.
+###############################################################################
+has 'qc_out'     => (is         => 'ro',
+                     isa        => 'NpgTrackingDirectory',
+                     required   => 1,);
+
+has 'rna_seqc_path' => (is         => 'ro',
+                       isa        => 'Str',
+                       lazy_build => 1,);
+
+sub _build_rna_seqc_path {
```

Comment:
If one considers this code outside of the pipeline framework, it is not clear why we are creating an extra directory level. If the unique path is needed, it should be supplied by the pipeline. The DOC for the check should explain what files are cteated and stress that the output files always have the same names. Then it's up to the caller how to deal with this. I believe we discussed this some time ago.